project_wbs/models/project_project.py

```diff
@@ -0,0 +1,342 @@
+                else:
+                    data.insert(0, '')
+                proj = proj.parent_id
+            data = '/'.join(data)
+            res2 = project_item.code_get()
+            if res2 and res2[0][1]:
+                data = '[' + res2[0][1] + '] ' + data
+
+            res.append((project_item.id, data))
+        return res
+
+    @api.multi
+    @api.depends('code')
+    def code_get(self):
+        res = []
+        for project_item in self:
+            data = []
+            proj = project_item
+            while proj:
+                if proj.code:
+                    data.insert(0, proj.code)
+                else:
+                    data.insert(0, '')
+
+                proj = proj.parent_id
+
+            data = '/'.join(data)
+            res.append((project_item.id, data))
+        return res
+
+    @api.multi
+    @api.depends('parent_id')
+    def _child_compute(self):
+        for project_item in self:
+            child_ids = self.search(
+                [('parent_id', '=', project_item.analytic_account_id.id)]
+            )
+            project_item.project_child_complete_ids = child_ids
+
+    @api.multi
+    def _get_parent_members(self):
+        context = self.env.context or {}
+        member_ids = []
+        project_obj = self.env['project.project']
+        if 'default_parent_id' in context and context['default_parent_id']:
+            for project in project_obj.search(
+                [('analytic_account_id', '=', context['default_parent_id'])]
+            ):
+                for member in project.members:
+                    member_ids.append(member.id)
+        return member_ids
+
+    @api.multi
+    def _get_analytic_complete_wbs_code(self):
+        for project in self:
+            project.c_wbs_code = \
+                project.analytic_account_id.complete_wbs_code
+
+    @api.multi
+    def _complete_wbs_code_search_analytic(self):
+        """ Finds projects for an analytic account.
+        @return: List of ids
+        """
+        project_ids = self.search([('analytic_account_id', 'in', self.ids)])
+        return project_ids
+
+    project_child_complete_ids = fields.Many2many(
+        'project.project',
+        string="Project Hierarchy",
+        compute=_child_compute
+    )
+    c_wbs_code = fields.Char(
+        compute=_get_analytic_complete_wbs_code,
+        string='WBS Code',
+        readonly=True,
+        store=True
+    )
+
+    members = fields.Many2many('res.users', 'project_user_rel', 'project_id',
+                               'uid', 'Project Members', help="""Project's
+                               members are users who can have an access to
+                               the tasks related to this project.""",
+                               default=_get_parent_members,
+                               states={'close': [('readonly', True)],
+                                       'cancelled': [('readonly', True)]})
```

Comment:
I'm +1 for this "members" feature, but I think it shouldn't be bundled here - it should go into it's own module.